sstables/sstables.hh

```diff
@@ -968,6 +969,10 @@ public:
         return _origin;
     }
 
+    const generation_type& original_sstable_generation() const noexcept {
```

Comment:
Shouldn't this be optional or something? Not all sstables will have this.

Reply:
The default constructed generation_type evaluates to `false`, so it will be engaged only if it was loaded from `scylla_metadata`

Reply:
This sounds like good opportunity for std::optimized_optional<> , it will make it immediately obvious that this generation can validly be missing

Reply:
Agree, in general we encourage using `optional<>` even when the type has a distinct "empty" state. Using `optional<>` wrapper makes the intended interface clear.

Reply:
Done in v4 ([633ca64](https://github.com/scylladb/scylladb/commit/633ca645486fa7314e5a1dd73c31141524cb4552))